subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -1432,7 +1507,214 @@ static inline u32_t ticker_job_insert(struct ticker_instance *instance,
+					    ext_data->ticks_drift;
+		} else {
+			/* Window has been exhausted - we can't reschedule */
+			ticker_id_iter = TICKER_NULL;
+		}
+
+		/* Try to find available slot for re-scheduling */
+		while (ticker_id_iter != TICKER_NULL &&
+		       ticks_start_offset + ticker->ticks_slot <=
+		       ticks_slot_window) {
+
+			node = &nodes[ticker_id_iter];
+
+			if (node->ext_data &&
+			    node->ext_data->reschedule_state ==
+			    TICKER_RESCHEDULE_STATE_PENDING) {
+				/* Skip other pending re-schedule nodes */
+				ticker_id_iter = node->next;
+				continue;
+			}
+
+			/* Calculate end of window. Since window may be aligned
+			 * with expiry of next node, we add a margin
+			 */
+			if (node->ticks_to_expire >
+			    HAL_TICKER_RESCHEDULE_MARGIN) {
+				window_end_ticks =
+					MIN(ticks_slot_window,
+					    ticks_start_offset +
+					    node->ticks_to_expire -
+					    HAL_TICKER_RESCHEDULE_MARGIN);
+			} else {
+				/* Next expiry is too close - try the next
+				 * node
+				 */
+				window_end_ticks = 0;
+			}
+
+			/* Calculate new ticks_to_expire as end of window minus
+			 * slot size => place at end of window.
+			 */
+			if (window_end_ticks > ticks_start_offset +
+			    ticker->ticks_slot) {
+				ticks_to_expire = window_end_ticks -
+					ticker->ticks_slot;
+			} else {
+				/* No space in window - try the next node */
+				ticks_to_expire = 0;
+			}
+
+			/* Decide if the re-scheduling ticker node fits in the
+			 * slot found - break if it fits
+			 */
+			if (ticks_to_expire != 0U &&
+			    ticks_to_expire >= window_start_ticks &&
+			    ticks_to_expire <= window_end_ticks -
+			    ticker->ticks_slot) {
+				/* Re-scheduled node fits before this node */
+				break;
+			}
+
+			/* We din't find a valid slot for re-scheduling - try
+			 * the next node
+			 */
+			ticks_start_offset += node->ticks_to_expire;
+			window_start_ticks  = node->ticks_to_expire +
+					      node->ticks_slot;
+			ticks_to_expire     = ticks_slot_window -
+					      ticker->ticks_slot;
+
+			ticker_id_prev = ticker_id_iter;
+			ticker_id_iter = node->next;
+		}
+
+		ticker->ticks_to_expire = ticks_to_expire;
+		ticker_id_iter = nodes[ticker_id_head].next;
+		ticker_id_prev = TICKER_NULL;
+
+		/* Place the ticker node sorted by expiration time and adjust
+		 * delta times
+		 */
+		while (ticker_id_iter != TICKER_NULL) {
+			node = &nodes[ticker_id_iter];
+			if (ticker->ticks_to_expire > node->ticks_to_expire) {
+				/* Node is after this - adjust delta */
+				ticker->ticks_to_expire -=
+					node->ticks_to_expire;
+			} else {
+				/* Node is before this one */
+				node->ticks_to_expire -=
+					ticker->ticks_to_expire;
+				break;
+			}
+			ticker_id_prev = ticker_id_iter;
+			ticker_id_iter = node->next;
+		}
+
+		if (ticker_id_prev != TICKER_NULL) {
+			/* Node did not become the first - insert it */
+			instance->ticker_id_head   = nodes[ticker_id_head].next;
+			nodes[ticker_id_head].next = nodes[ticker_id_prev].next;
```

Comment:
Action: Add blank line to separate head update with re-linking. Second part comment: "Re-insert/link ticker nodes"